packages/react/src/index.d.ts

```diff
@@ -4,6 +4,9 @@ declare const UNDEFINED_VOID_ONLY: unique symbol;
 // Destructors are only allowed to return void.
 type Destructor = () => void | { [UNDEFINED_VOID_ONLY]: never };
 
+type InferNone<T> = T extends undefined ? typeof React.None : never;
+type MapToNone<T> = { [K in keyof T]-?: NonNullable<T[K]> | InferNone<T[K]> };
```

Comment:
This type doesn't seem to require the required modifier `-?:` since [`setState` accepts partial state objects](https://legacy.reactjs.org/docs/state-and-lifecycle.html#state-updates-are-merged). I think that modifier can be removed.